internal/store/horizontalpodautoscaler.go

```diff
@@ -56,6 +56,23 @@ var (
 
 func hpaMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generator.FamilyGenerator {
 	return []generator.FamilyGenerator{
+		*generator.NewFamilyGenerator(
+			"kube_horizontalpodautoscaler_info",
+			"Information about this autoscaler.",
+			metric.Gauge,
+			"",
+			wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelKeys:   []string{"scaletargetref_api_version", "scaletargetref_kind", "scaletargetref_name"},
+							LabelValues: []string{a.Spec.ScaleTargetRef.APIVersion, a.Spec.ScaleTargetRef.Kind, a.Spec.ScaleTargetRef.Name},
```

Comment:
Can we add a test to see what happens when a.Spec.ScaleTargetRef is not set?

Reply:
The OpenAPI specification tells us that `a.Spec.ScaleTargetRef` is required.
https://github.com/kubernetes/kubernetes/blob/0d3f2ca371d03cc58c36d3ee3523b719e987fc35/api/openapi-spec/v3/apis__autoscaling__v2beta2_openapi.json#L323-L326

Though the `CrossVersionObjectReference` does not require the `apiVersion` property and thus could be nil.
https://github.com/kubernetes/kubernetes/blob/0d3f2ca371d03cc58c36d3ee3523b719e987fc35/api/openapi-spec/v3/apis__autoscaling__v2beta2_openapi.json#L74-L77

Agree to add validation for `a.Spec.ScaleTargetRef.APIVersion` only?

Reply:
Thanks for double checking, I remember that we had a similar issue for VPA. In this case, let's just add a test for the APIVersion field